File: drills/ts/harnesses/nc73_word_break.ts
import * as solution from "./solution";

const show = (v: unknown) => JSON.stringify(v) ?? "undefined";

export function run(): [string, string, string][] {
  if (typeof solution.wordBreak !== "function") throw new Error("__signature_mismatch__");
  return [
    ["wordBreak('leetcode', ['leet', 'code'])", show(true), show(solution.wordBreak('leetcode', ['leet', 'code']))],
    ["wordBreak('applepenapple', ['apple', 'pen'])", show(true), show(solution.wordBreak('applepenapple', ['apple', 'pen']))],
    ["wordBreak('catsandog', ['cats', 'dog', 'sand', 'and', 'cat'])", show(false), show(solution.wordBreak('catsandog', ['cats', 'dog', 'sand', 'and', 'cat']))],
    ["wordBreak('', ['a'])", show(true), show(solution.wordBreak('', ['a']))],
    ["wordBreak('a', [])", show(false), show(solution.wordBreak('a', []))],
    ["wordBreak('aaaaaaa', ['aaa', 'aaaa'])", show(true), show(solution.wordBreak('aaaaaaa', ['aaa', 'aaaa']))],
  ];
}

File: drills/go/harnesses/nc73_word_break.go
package main

func main() {
	run(func() []testCase {
		return []testCase{
			tc("wordBreak('leetcode', ['leet','code'])", true, wordBreak("leetcode", []string{"leet", "code"})),
			tc("wordBreak('applepenapple', ['apple','pen'])", true, wordBreak("applepenapple", []string{"apple", "pen"})),
			tc("wordBreak('catsandog', ['cats','dog','sand','and','cat'])", false, wordBreak("catsandog", []string{"cats", "dog", "sand", "and", "cat"})),
			tc("wordBreak('', ['a'])", true, wordBreak("", []string{"a"})),
			tc("wordBreak('a', [])", false, wordBreak("a", []string{})),
			tc("wordBreak('aaaaaaa', ['aaa','aaaa'])", true, wordBreak("aaaaaaa", []string{"aaa", "aaaa"})),
		}
	})
}
